coll/jq.go

```diff
@@ -0,0 +1,36 @@
+package coll
+
+import (
+	"github.com/itchyny/gojq"
+	"github.com/pkg/errors"
+)
+
+// JQ -
+func JQ(jqExpr string, in interface{}) (interface{}, error) {
+	query, err := gojq.Parse(jqExpr)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse JQ %s: %w", jqExpr, err)
+	}
+	iter := query.Run(in)
+	var out interface{}
+	a := []interface{}{}
+	for {
+		v, ok := iter.Next()
+		if !ok {
+			break
+		}
+		if err, ok := v.(error); ok {
+			return nil, fmt.Errorf("executing JQ failed: %w", err)
+		}
+		if v != nil { // TODO: Check, if nil may be a valid result
+			a = append(a, v)
+		}
+	}
+	if len(a) == 1 {
+		out = a[0]
+	} else {
+		out = a
+	}
```

Comment:
@hairyhenderson I'm not sure if it is a good idea to magically turn an array with one result into a single item as it is done for `jsonpath` and (since it was my template for jq) is currently also done for jq the same way. If an array returns just one element by accident it may break the template.
Or do I miss something?

Reply:
It seems, that there's not an easy way to distinguish between an array with a single element and just an element in the result. Thus I keep it the same way as it is done for `jsonpath`.